app/src/main/java/space/taran/arknavigator/mvp/presenter/SettingsPresenter.kt

```diff
@@ -0,0 +1,74 @@
+package space.taran.arknavigator.mvp.presenter
+
+import android.util.Log
+import kotlinx.coroutines.launch
+import moxy.MvpPresenter
+import moxy.presenterScope
+import ru.terrakok.cicerone.Router
+import space.taran.arknavigator.mvp.model.UserPreferences
+import space.taran.arknavigator.mvp.model.UserPreferences.*
+import space.taran.arknavigator.mvp.view.SettingsView
+import space.taran.arknavigator.utils.SETTINGS_SCREEN
+import javax.inject.Inject
+
+class SettingsPresenter : MvpPresenter<SettingsView>() {
+
+    @Inject
+    lateinit var router: Router
+
+    @Inject
+    lateinit var userPreferences: UserPreferences
+
+    override fun onFirstViewAttach() {
+        Log.d(SETTINGS_SCREEN, "first view attached in SettingsPresenter")
+        super.onFirstViewAttach()
+
+        viewState.init()
+        presenterScope.launch {
+            notifyAllPreferences()
+        }
+    }
+
+    fun onCrashPreferenceClick(crashReport: CrashReport) {
```

Comment:
I would name it `onCrashReportingClick` to be consistent with the rest of methods.